docs/extend-graphql-schema.md

```diff
@@ -3,4 +3,568 @@ id: extend-graphql-schema
 title: Extend the GraphQL Schema
 ---
 
-How to extend the existing GraphQL Schema?
\ No newline at end of file
+How to extend the existing GraphQL Schema?
+
+You might at some point need to expose new data in your GraphQL endpoint, to
+support new features. Front-Commerce allows you to extend and override any part
+of the initial schema by leveraging features from the GraphQL Schema Definition
+Language (<abbr title="Schema Definition Language">SDL</abbr>).
+
+This can be achieved by creating and registering your own custom « modules » on
+the server. Front-Commerce Lite uses [GrAMPS](https://gramps.js.org/)
+datasources, and Front-Commerce uses a similar approach with additional
+features.
+
+## Adding a custom module
+
+Let’s say that we would like to expose a list of physical stores in our GraphQL
+schema. We first have to create a `PhysicalStores` module.
+
+To do so, create its definition as follow in the
+`src/server/modules/physical-stores/index.js` file:
+
+```js
+export default {
+  namespace: "PhysicalStores"
+};
+```
+
+The module does nothing yet, but you can now register it in the application by
+updating `src/server/modules.js` as follow:
+
+```diff
+import StoreInformation from "./modules/store-information";
++import PhysicalStores from "./modules/physical-stores";
+
+export default [
+  // In a real Front-Commerce application you could load core modules
+  // on demand right here, and select the parts you need for your current
+  // project.
+  // This is where you could combine several core datasources to build
+  // the project that match your needs and run away from a monolithic
+  // architecture when it is not relevant
+
+-  StoreInformation
++  StoreInformation,
++  PhysicalStores
+];
+```
+
+Congratulations! You now have a new custom module ready to enhance the data
+exposed in the GraphQL middleware.
+
+## Extend the GraphQL schema
+
+All the wiring is now done and it is time to develop the features of this
+module. In GraphQL, a good practice is to start thinking about the schema first
+from a business domain standpoint.
+
+It is important to name things with a language shared by the team and prevent
+exposing implementation details (ids, different names…) as much as possible. We
```

Comment:
While I understand your point for `ids` it's definitly not true from Apollo's point of view since it helps for automatic caching strategies.

I'd rather mention the fact that you don't want a `relation_id` field but a `relation` field which will be a node containing all the data of the relevant `relation`.